lib/faktory_worker/job.ex

```diff
@@ -0,0 +1,81 @@
+defmodule FaktoryWorker.Job do
+  @moduledoc """
+  todo: docs
+  """
+
+  # Look at supporting the following optional fields when pushing a job
+  # priority
+  # reserve_for
+  # at
+  # retry
+  # backtrace
+  # created_at
+  # custom
+  @optional_job_fields [:queue]
+
+  defmacro __using__(using_opts \\ []) do
+    alias FaktoryWorker.Job
+
+    quote do
+      def perform_async(job, opts \\ []) do
+        opts = Keyword.merge(unquote(using_opts), opts)
+
+        __MODULE__
+        |> Job.create_job(job, opts)
+        |> Job.perform_async(opts)
+      end
+    end
+  end
+
+  @doc false
+  def create_job(worker_module, job, opts) when is_list(job) do
```

Comment:
Should the `job` param be called `args` here?

Reply:
Also `create_job` is not very elixir-y 😄 what would be a better fit for this function name?

Reply:
Yep we can call it `args` instead as that makes much more sense.

I guess the function name would be `Job.new` if we were to follow the general conventions. I did think about using a struct here as well so that's open to discussion.

Reply:
What about naming the function something like `Job.payload()` or `Job.build_payload()` so it matches what Faktory calls it? https://github.com/contribsys/faktory/wiki/The-Job-Payload

Reply:
Good shout, I will make that update.